File: browserJava/src/main/java/com/luciasoft/browserjava/multibrowser/MyListDialog.java
package com.luciasoft.browserjava.multibrowser;

import android.content.Context;
import android.content.DialogInterface;

import androidx.appcompat.app.AlertDialog;

class MyListDialog
{
    private int choice = -1;

    private void setChoice(int choice)
    {
        this.choice = choice;
    }

    int getChoice()
    {
        return choice;
    }

    void show(Context context, String title, final String[] options, int defaultChoice, DialogInterface.OnClickListener onOkClick)
    {
        if (title == null) title = "";

        AlertDialog.Builder builder = new AlertDialog.Builder(context);

        builder.setTitle(title);

        builder.setSingleChoiceItems(options, defaultChoice, new DialogInterface.OnClickListener()
        {
            @Override
            public void onClick(DialogInterface dialog, int which)
            {
                setChoice(which);
            }
        });

        builder.setPositiveButton("OK", onOkClick);

        builder.setNegativeButton("Cancel", null);

        AlertDialog dialog = builder.create();

        dialog.show();
    }
}

File: tempFiles/browserJavaToKotlin - Copy/src/main/java/com/luciasoft/browserjavatokotlin/MyListDialog.kt
package com.luciasoft.browserjavatokotlin

import android.content.Context
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog

internal class MyListDialog
{
    var choice = -1
        private set

    fun show(
        context: Context,
        title: String = "",
        options: Array<String>,
        defaultChoice: Int,
        onOkClick: DialogInterface.OnClickListener? = null
    )
    {
        with (AlertDialog.Builder(context))
        {
            this.setTitle(title)
            this.setSingleChoiceItems(
                options,
                defaultChoice,
                object : DialogInterface.OnClickListener
                {
                    override fun onClick(dialog: DialogInterface, which: Int)
                    {
                        choice = which
                    }
                })
            this.setPositiveButton("OK", onOkClick)
            this.setNegativeButton("Cancel", null)
            this.create().show()
        }
    }
}